jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/AppServerAutoDetectionTest.java

```diff
@@ -121,28 +122,28 @@ public void testDefaultServer() throws IOException {
             .build()).build();
 
         AppServerHandler appServerHandler = new AppServerDetector(generatorContext).detect(null);
-        assertEquals("tomcat", appServerHandler.getName());
+        assertThat(appServerHandler.getName()).isEqualTo("tomcat");
     }
 
     private void assertAppServerDescriptorApplicability(Object[] descriptors) throws IOException {
```

Comment:
Maybe this method needs some more explanation (in a comment?). I read it several times but still don't know what it is doing.

Reply:
Here we are checking that if the app server is null or not specified, we should get the default server, which is tomcat.

Maybe this method name will explain what it is doing.
```java
detect_withNullServer_shouldReturnTomcatAsDefaultServer()
```

Reply:
I was refering to `assertAppServerDescriptorApplicability`

Reply:
OK I understand now ... what is confusing is that we are building the tuples (parameter/expectation) that are then extracted to be asserted ... Is it possible to make it simpler to implement and read.

Reply:
I tried to make it simple like this
```java
private void assertAppServerDescriptorApplicability(String appDir, String file, boolean expected) throws IOException {
        Random random = new Random();
        int i = random.nextInt(1000);

        Path webInf = Files.createDirectories(folder.resolve("webapp" + i).resolve(appDir));
        Files.createFile(webInf.resolve(file));

        GeneratorContext generatorContext = GeneratorContext.builder().project(JavaProject.builder()
                .buildDirectory(webInf.toFile())
                .plugins(Collections.emptyList())
                .build()).build();

        boolean result = new AppServerDetector(generatorContext).detect(null).isApplicable();
        assertThat(result).isEqualTo(expected);
}
```
but due to this statement
https://github.com/eclipse/jkube/blob/40ce95a6366986bd005b51420a799f09b5e075f8/jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/AppServerAutoDetectionTest.java#L132

we need a random integer or we can change signature of `assertAppServerDescriptorApplicability` method to accept an additional argument.

Reply:
We can simplify `assertAppServerDescriptorApplicability` method like this

```java
private void assertAppServerDescriptorApplicability(String appDir, String file, boolean expected) throws IOException {
    File webInf = Files.createDirectories(folder.resolve("webapp").resolve(appDir)).toFile();
    Files.createFile(webInf.toPath().resolve(file));
    try {
      GeneratorContext generatorContext = GeneratorContext.builder().project(JavaProject.builder()
          .buildDirectory(webInf)
          .plugins(Collections.emptyList())
          .build()).build();

      boolean result = new AppServerDetector(generatorContext).detect(null).isApplicable();
      assertThat(result).isEqualTo(expected);
    } finally {
      FileUtils.deleteDirectory(webInf);
    }
  }
```
will this work?

Reply:
yes it is better

Reply:
then, If it has to be in another PR it works for me.

Reply:
Sure, I see https://github.com/eclipse/jkube/issues/1662 that's why I've not refactored it in this PR.